### PR TITLE
Adds terraform skeleton and GitHub code

### PR DESCRIPTION
**Pull request by KoeSystems**
This PR adds the bases for terraform use, the Makefile, directory tree, providers, etc...
As the first code adds GitHub code for organizations and repositories, all this code has been tested successfully with terraform import

For now, we don't have the backend.tf, so please do not apply this code, but you can test it with `make plan` and terraform will say it wants to create all the resources despite are already created.

I will manage this when we have tf backend set in place.

### CODE REVIEW

File: terraform/gcp/01-gke.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+resource "google_container_cluster" "primary" {
+  name                     = "bcneng"
+  location                 = "europe-west1-b"
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  master_auth {
+    username = ""
+    password = ""
```
Comment (gonzaloserrano) [inline terraform/gcp/01-gke.tf:9]: this means that the bucket is public, right?

File: terraform/github/Makefile [hunk 2]
```diff
@@ -0,0 +1 @@
+../Makefile
```
Comment (gonzaloserrano) [inline terraform/github/Makefile:1]: Just wondering, what's this for?

File: terraform/gcp/01-gke.tf (hunk 1 above)
Comment (KoeSystems) [reply terraform/gcp/01-gke.tf:9]: is not real code for GKE, I'm working in a real tf code for our cluster...

Comment (KoeSystems) [reply terraform/gcp/01-gke.tf]: removed from the PR for now.

File: terraform/github/01-repositories.tf [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  description            = "Backend for salary tracker from the Barcelona area"
+  allow_merge_commit     = false
+  allow_rebase_merge     = true
+  allow_squash_merge     = true
+  has_downloads          = true
+  has_issues             = true
+  has_projects           = true
+  has_wiki               = true
+  delete_branch_on_merge = true
+}
```
Comment (gonzaloserrano) [inline terraform/github/01-repositories.tf:88]: I think an EOF is missng here

File: terraform/github/00-organization.tf [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  repository = "meetups"
+  username   = "ignasi35"
+  permission = "push"
+}
+
+resource "github_repository_collaborator" "ruben-perez_meetups" {
+  repository = "meetups"
+  username   = "ruben-perez"
+  permission = "push"
+}
```
Comment (gonzaloserrano) [inline terraform/github/00-organization.tf:94]: probably should migrate also to members plus a new group... but not part of this PR

File: terraform/Makefile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+__check_defined = \
+		$(if $(value $1),, \
+			$(error Undefined $1$(if $2, ($2))))
+
+help:
+		@echo
+		@echo "Please use make <option> where <option> is one of:"
+		@echo
+		@echo "plan     "
+		@echo "apply    "
```
Comment (gonzaloserrano) [inline terraform/Makefile:25]: I don't see the `apply` rule 🤔

File: terraform/github/Makefile (hunk 2 above)
Comment (KoeSystems) [reply terraform/github/Makefile:1]: Makefile lives in `terraform` path, but we will have multiple providers, github, gcp, ns1, etc... so with a symbolic link we can reuse the Makefile.

File: terraform/Makefile (hunk 5 above)
Comment (KoeSystems) [reply terraform/Makefile:25]: not yet :)

**Comment by gonzaloserrano** [comment]
💃
